Add Azure AD capability to module

madmatt opened the pull request:
This is the beginnings of a PR to add Azure AD capability to the module. Still need to determine if it's backwards compatible (I think it is).

It has been tested against a stock standard Azure AD trial account and works well, provided the new instructions are followed by the Azure AD administrator.

madmatt commented:
Have just removed the `[WIP]` tag, this is being used in a few different production sites now and seems to work well across environments. I haven't yet been able to review these changes against an ADFS target to make sure nothing has regressed, and I don't expect to have the opportunity to do this anytime soon unfortunately.

Perhaps we can release this as a new major version so it doesn't break existing installs that run `composer update`.

This can now be reviewed.

NightJar commented on src/Services/SAMLConfiguration.php:
```diff
@@ -52,6 +52,25 @@ class SAMLConfiguration
+     *
+     * Defaults to true to preserve backwards compatibility (ADFS).
+     */
+    private static $expect_binary_nameid;
```
Docblock states that this value defaults to true, however there is no default value, thus it is false (`NULL`).
If it is true from yml sources I think it best also reflected here in the "hard truth", coinciding with the docblock.

madmatt commented:
Thanks @NightJar, I've fixed that up now, and also added some extra functionality. Will fix the linting errors at some point shortly, but could you possible review my changes above? I'll squash before merge, but figured you'd prefer to only look at the new commits for now ;-)

madmatt commented:
Without spending a lot of time faking SAML responses I don't see a nice / easy way to test the SAMLController->acs() method, and that's outta scope for me right now. Other than that issue with codecov.io, I think this is ready for merge.

@NightJar Any chance of a quick review of the latest commits, and if you're happy I'll squash and you can merge, then I can release? Thanks!
